TestBenches/VMRouter_test.cpp

```diff
@@ -15,245 +14,112 @@ const int nEvents = 100;  //number of events to run
 // Sort stubs into smaller regions in phi, i.e. Virtual Modules (VMs).
 
 // NOTE: to run a different phi region, change the following
-//          - the included top function in VMRouter_test.cpp (if file name is changed)
-//          - the top function and memory directory in script_VMR.tcl (if file name is changed)
+//          - constants specified in VMRouterTop.h
+//          - add number to VMRouter_parameters.h if not already defined
 //          - add the phi region in emData/download.sh, make sure to also run clean
-//          - and the changes listed in VMRouterTop.cc/h
+//          - add/remove pragmas depending on inputStubs in VMRouterTop.cc (not necessary to run simulation)
 
 
-// Finds all memory names for the specified processing module found in the wiring file,
-// and adds them to the fileNames array with the memory files directory, excluding the copies nX.
-// The number of copies are kept track in the numCopiesArray.
-// E.g. finds all memories that start with "VMSME_L1PHIE", such as "VMSME_L1PHIE17" etc.
-// Returns false if wiring file isn't found.
-template<int arraySize>
-bool findFileNames(string fileDirStart, string wireFileName, string memID, string nameList[arraySize], int numCopiesArray[arraySize]) {
-
-  ifstream wireFile; // Will contain the wiring file
-
-  int nvmMemories = 0; // Number of memories found
-  char delimeter = (memID.substr(0,2) == "IL") ? ' ' : 'n'; // Different delimeter if input or output memories
-
-  bool valid = openDataFile(wireFile, wireFileName); // Open the wiring file
-
-  // Check if the wiring file was opened properly.
-  if (not valid) {
-    cout << "Could not find wiring file." << endl;
-    return false;
+// Count the number of copies of each memory in a file name vector
+vector<int> countCopies(vector<string> fileNames) {
+  vector<int> numCopies; // Each element in vector corresponds to one memory, i.e. L1PHIE17
+  string firstCopyIndex = fileNames.begin()->substr(fileNames.begin()->find("n")+1, 1); // The number after n
+  for (auto f = fileNames.begin(); f != fileNames.end(); ++f) {
+    if (f->find("n"+firstCopyIndex) != string::npos) numCopies.push_back(1);
+    else numCopies.back() += 1;
```

Comment:
Could this function be useful in any other test benches? If so, you could convert it to a method of the `TBHelper` class.

Reply:
I don't think so. I need it because I use two-dimensional arrays of memories, but no other module uses that. And what those two dimensions correspond to seems quite specific to the VMR.

Reply:
OK, makes sense. Then could you please change `vector<string> fileNames` to `const vector<string> &fileNames`? That will prevent an unnecessary copy of the input vector.